### PR TITLE
Basic structure without styling and state

### PR DESCRIPTION
**Pull request by IlyaPapou**
All comments would be welcome

### CODE REVIEW

File: src/components/weatherInfoContainer/weatherInfoContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    );
+  }
+
+  render() {
+    const {
+      weatherIcon, time, cityName, temperature,
+    } = this.state;
+    return (
+      <WeatherInfo
+        cityName={cityName}
```
Comment (AV29) [inline src/components/weatherInfoContainer/weatherInfoContainer.jsx]: OPTIONAL... could also do <WeatherInfo {...this.state }/>
...but there should be a balance for better expressiveness... - let's leave as is... what do you think?

File: src/components/weatherPreviewList/weatherPreviewList.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React, { Component } from 'react';
+import WeatherPreviewRow from '../weatherPreviewRow/weatherPreviewRow.jsx';
+
+// eslint-disable-next-line no-unused-vars
+const GetCitiesListWithWeather = () => new Promise((resolve, reject) => {
```
Comment (AV29) [inline src/components/weatherPreviewList/weatherPreviewList.jsx]: camelCase ?

File: src/components/weatherInfoContainer/weatherInfoContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import WeatherInfo from '../weatherInfo/weatherInfo.jsx';
```
Comment (AV29) [inline src/components/weatherInfoContainer/weatherInfoContainer.jsx]: files with components are usually CapitalCased

File: src/components/mainScreen/mainScreen.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          logo,
+          homeIcon,
+          isMore,
+          homeUrl,
+        } = this.state;
+        return (
+          <>
+            <Header logo={logo} />
+            <Sidebar homeUrl={homeUrl} homeIcon={homeIcon} />
+            <>
```
Comment (AV29) [inline src/components/mainScreen/mainScreen.jsx]: you don't need to use fragments here

File: src/components/header/header.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Header = ({ logo }) => (
+  <div>
```
[the remark this answers is not in the document]
Comment (IlyaPapou) [reply src/components/header/header.jsx]: yep, done

File: src/components/mainScreen/mainScreen.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        source: 'smth',
+        alt: 'smth',
+      },
+      homeIcon: {
+        source: 'smth',
+        alt: 'smth',
+      },
+      isMore: false,
+      homeUrl: 'Smth',
+
```
[the remark this answers is not in the document]
Comment (IlyaPapou) [reply src/components/mainScreen/mainScreen.jsx]: yep, redundant space deleted

File: src/components/mainScreen/mainScreen.jsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+          homeUrl,
+        } = this.state;
+        return (
+          <>
+            <Header logo={logo} />
+            <Sidebar homeUrl={homeUrl} homeIcon={homeIcon} />
+            <>
+              {isMore ? <WeatherInfoContainer /> : <WeatherPreviewList />}
+            </>
+            <Button text={isMore ? 'More' : 'Back'} handleClick={this.onHandleClick} />
```
[the remark this answers is not in the document]
Comment (IlyaPapou) [reply src/components/mainScreen/mainScreen.jsx]: roger that

File: src/components/sidebar/sidebar.jsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Sidebar = ({ homeUrl, homeIcon }) => (
+  <div>
```
[the remark this answers is not in the document]
Comment (IlyaPapou) [reply src/components/sidebar/sidebar.jsx]: yes, we could

File: src/components/weatherInfoContainer/weatherInfoContainer.jsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+// eslint-disable-next-line no-unused-vars
+const GetCityWeatherInfo = () => new Promise((resolve, reject) => {
```
[the remark this answers is not in the document]
Comment (IlyaPapou) [reply src/components/weatherInfoContainer/weatherInfoContainer.jsx]: roger that

File: src/components/weatherInfoContainer/weatherInfoContainer.jsx (hunk 1 above)
Comment (IlyaPapou) [reply src/components/weatherInfoContainer/weatherInfoContainer.jsx]: I guess it's better to leave as is

File: src/components/weatherPreviewList/weatherPreviewList.jsx (hunk 2 above)
Comment (IlyaPapou) [reply src/components/weatherPreviewList/weatherPreviewList.jsx]: +

File: src/components/weatherInfoContainer/weatherInfoContainer.jsx (hunk 3 above)
Comment (IlyaPapou) [reply src/components/weatherInfoContainer/weatherInfoContainer.jsx]: +

File: src/components/button/button.jsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const ButtonView = styled.button`
```
Comment (ximet) [inline src/components/button/button.jsx]: better move styles to another file like Button.style.js

File: src/components/header/header.jsx [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import styled from 'styled-components';
+import Clock from '../clock/clock';
+
+const HeaderWrapper = styled.div`
```
Comment (ximet) [inline src/components/header/header.jsx]: the same

File: src/store.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+const configureStore = (preloadedState) => (
+  createStore(
+    rootReducer,
+    preloadedState,
+    composeEnhancers(applyMiddleware(thunk)),
+  )
+);
+
+const store = configureStore({});
```
Comment (ximet) [inline src/store.js]: better use initial state, not empty object

File: src/store/actions/actionCreator.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import { ACTION_TYPES } from '../../constants';
+import WeatherService from '../../services/weather.service';
+import { TransformCityWeather } from '../../utils/transformer';
+
+
+export const fetchWeatherByCityAction = (city) => (async (dispatch) => {
+  try {
+    const weather = await WeatherService.getCurrentWeatherByCity(city);
+    dispatch({
```
Comment (ximet) [inline src/store/actions/actionCreator.js]: better create action creator function

File: src/utils/transformer.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+export const kelvinToCelsius = (kelvin) => (kelvin - 273.15);
+
+export const checkForEmptyObject = (object) => !(Object.entries(object).length === 0 && object.constructor === Object);
+
+export const capitalizeFirstLetter = (string) => string.charAt(0).toUpperCase() + string.substring(1);
+
+export const TransformCityWeather = (weatherInfo) => {
+  if (checkForEmptyObject(weatherInfo)) {
+    return {
+      name: weatherInfo.name,
```
Comment (ximet) [inline src/utils/transformer.js]: what if some param inside weatherInfo undefined?

File: src/сontainers/App/App.jsx [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: baseline;
+  margin: 40px 40px 40px 80px;
+`;
+
+class App extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (ximet) [inline src/сontainers/App/App.jsx]: you can use redux, without the local state. OR you can use hooks

File: src/сontainers/WeatherPreviewRowsContainer/WeatherPreviewRowsContainer.jsx [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+  selectedCity: state.setSelectedCity,
+});
+
+const mapDispatchToProps = (dispatch) => ({
+  fetchWeatherByCity: (cityName) => dispatch(fetchWeatherByCityAction(cityName)),
+  setSelectedCity: (city) => dispatch(setSelectedCityAction(city)),
+});
+
+WeatherPreviewRowsContainer.propTypes = {
+  fetchWeatherByCity: PropTypes.func.isRequired,
```
Comment (ximet) [inline src/сontainers/WeatherPreviewRowsContainer/WeatherPreviewRowsContainer.jsx]: every param required?

**Pull request closed without merging**